rbac/management/audit_log/model.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""Model for audit logging."""
+from django.db import models
+from django.utils import timezone
+
+from api.models import Tenant, TenantAwareModel
+
+
+class AuditLog(TenantAwareModel):
+    """An audit log."""
+
+    GROUP = "group"
+    ROLE = "role"
+    USER = "user"
+    PERMISSION = "permission"
+    RESOURCE_CHOICES = (
+        (GROUP, "Group"),
+        (ROLE, "Role"),
+        (USER, "User"),
+        (PERMISSION, "Permission"),
+    )
+
+    DELETE = "delete"
+    ADD = "add"
+    EDIT = "edit"
+    CREATE = "create"
+    REMOVE = "remove"
+    ACTION_CHOICES = (
+        (DELETE, "Delete"),
+        (ADD, "Add"),
+        (EDIT, "Edit"),
+        (CREATE, "Create"),
+        (REMOVE, "Remove"),
+    )
+
+    created_at = models.DateTimeField(default=timezone.now)
+    requester = models.TextField(max_length=255, null=False)
+    description = models.TextField(max_length=255, null=False)
+    resource = models.CharField(max_length=32, choices=RESOURCE_CHOICES)
```

Comment:
as I am looking on https://issues.redhat.com/browse/RHCLOUD-19119 - we need track not only resource type(Group, Role,..) but also its ID.

this can solved by polymorphic relation but we don't have use it in our project yet. YOu can learn more [here](https://django-polymorphic.readthedocs.io/en/stable/quickstart.html).

But It looks that we don't need use library `django-polymorphic` as it would bring more unnecessary complexity for us.

We just need to store resource_type(from Group,.... ) and resource_id (it's id ) and it would be easy to work it later in app.

Let me know if I understand our use case differently.

Reply:
That makes sense!

Reply:
just the id for example group#id

Reply:
This PR has the code that I've been using to test my changes as I update the changes to this PR to see how it works, and I added a helper function to help find the resource_id if you'd like to take a look/more information and if you have any thoughts on that? 
https://github.com/RedHatInsights/insights-rbac/pull/980/files#diff-bc55ca994bea95eeff3fdd4be34ab5d19d0f33c5c04d985418aee68dfb35627eR69
Thank you so much!